x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -300,6 +301,8 @@ public static boolean canPushToSource(Expression exp, Predicate<FieldAttribute>
                     && Expressions.foldable(cidrMatch.matches());
             } else if (exp instanceof SpatialRelatesFunction bc) {
                 return bc.canPushToSource(LocalPhysicalPlanOptimizer::isAggregatable);
+            } else if (exp instanceof StringQueryPredicate) {
+                return true;
```

Comment:
Right. At the moment it's just an opaque string.

Reply:
You'd have to do that much earlier than the *Local* PhysicalPlanOptimizer I think.

Reply:
In ES|QL "retrieval" of fields and how they are resolved (what type they have, if the name is correct or not, if there are conflicts between data types or not) is done in a different layer.

Reply:
Given that we're still not retrieving the fields that are involved in the query string, is there something we need to do at this stage besides always trying to push it to source?

My thinking was to check that afterwards. using maybe the `LogicalVerifier` to double check that there were no `Filter` plans with unsupported `Expression` conditions. 

Would that approach work, or should any other approach work better for an (as of now) opaque query string?

Reply:
LogicalVerifier has a different role: it checks that every logicalPlan has the correct set of inputs and outputs (that it doesn't create un-tracked columns or loses references along the way).

You need a check similar to what Ioana did with the `match` operator in the Verifier.

Reply:
Thanks for clarifying @astefan !

I don't think we can provide something similar until the fields from the query string are parsed in follow up iterations. 

Do y'all think there's something specific we can do before that happens to prevent an invalid use of the command?

Reply:
Think about queries where `match` comes after some commands that you know for sure it cannot be ok. For example, after `stats` for sure it's not ok because stats aggregates the results; doing a `match` afterwards is pointless.

Reply:
Taking a closer look, I think there's only a couple of commands where it would be safe to use `MATCH` after (without field information): `WHERE` and `SORT`: ed1852ec42dfa6077b7c43ec066a76d92d93dd9d

We're very conservative there, but it's probably better that way until we can decide on optimizations and validations.

I've added some tests as well, they're incomplete - pending validation of this approach.

WDYAT?